src/main/java/io/quarkus/gizmo2/impl/BlockCreatorImpl.java

```diff
@@ -1048,6 +1049,28 @@ public Expr setOf(final List<Expr> items) {
         }
     }
 
+    @Override
+    public Expr mapOf(List<Expr> items) {
+        int size = items.size();
+        if (size % 2 != 0) {
+            throw new IllegalArgumentException("Invalid number of items: " + items);
+        }
+        if (size <= 10) {
+            List<Expr> args = new ArrayList<>(size * 2);
+            for (Expr item : items) {
+                args.add(item);
+            }
+            return invokeStatic(MethodDesc.of(Map.class, "of", Map.class, nCopies(args.size(), Object.class)), args);
+        } else {
+            int idx = 0;
+            Expr args = newEmptyArray(Util.classDesc(Entry.class), Constant.of(size));
+            for (Expr item : items) {
+                set(args.elem(idx++), item);
+            }
+            return invokeStatic(MethodDesc.of(Map.class, "ofEntries", Map.class, Entry[].class), newArray(Object.class, args));
```

Comment:
This part doesn't look right, the new array will be the wrong size and it will not contain map entries as is required by the `ofEntries` method.

This is actually a very tricky thing because each entry has to be constructed *before* the next entry expressions appear on the instruction list; i.e. calls to `Map.entry(k, v)` have to be stitched into the list after the fact. It is doable (this is similar to how `newArray(Expr...)` is implemented) but the algorithm is pretty complex.

Maybe it'd be best to throw an `UnsupportedOperationException` for maps with more than 10 entries for now, and we can revisit it in a second pass.

Reply:
Good catch! I actually had a TODO comment in my local branch but forgot to revisit this code.  BTW this part is not tested at all ;-).

> Maybe it'd be best to throw an UnsupportedOperationException for maps with more than 10 entries for now, and we can revisit it in a second pass.

+1

Reply:
Done.